internal/services/mobilenetwork/mobile_network_attached_data_network_resource.go

```diff
@@ -0,0 +1,617 @@
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.AttachedDataNetworkClient
+
+			id, err := attacheddatanetwork.ParseAttachedDataNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			if err := resourceMobileNetworkChildWaitForDeletion(ctx, id.ID(), func() (*http.Response, error) {
+				resp, err := client.Get(ctx, *id)
+				return resp.HttpResponse, err
+			}); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandNaptConfigurationModel(inputList []NaptConfigurationModel) *attacheddatanetwork.NaptConfiguration {
+	if len(inputList) == 0 {
+		return nil
+	}
+
+	input := &inputList[0]
+	output := attacheddatanetwork.NaptConfiguration{
+		PinholeLimits: &input.PinholeLimits,
+	}
+
+	naptEnabled := attacheddatanetwork.NaptEnabledDisabled
+	if input.Enabled {
+		naptEnabled = attacheddatanetwork.NaptEnabledEnabled
+	}
+	output.Enabled = &naptEnabled
+
+	output.PinholeTimeouts = expandPinholeTimeoutsModel(input.PinholeTimeouts)
+
+	output.PortRange = expandPortRangeModel(input.PortRange)
+
+	output.PortReuseHoldTime = expandPortReuseHoldTimesModel(input.PortReuseHoldTime)
+
+	return &output
+}
+
+func expandPinholeTimeoutsModel(inputList []PinholeTimeoutsModel) *attacheddatanetwork.PinholeTimeouts {
+	if len(inputList) == 0 {
+		return nil
+	}
+
+	input := &inputList[0]
+	output := attacheddatanetwork.PinholeTimeouts{
+		Icmp: &input.Icmp,
+		Tcp:  &input.Tcp,
+		Udp:  &input.Udp,
+	}
+
+	return &output
+}
+
+func expandPortRangeModel(inputList []PortRangeModel) *attacheddatanetwork.PortRange {
+	if len(inputList) == 0 {
+		return nil
+	}
+
+	input := &inputList[0]
+	output := attacheddatanetwork.PortRange{
+		MaxPort: &input.MaxPort,
+		MinPort: &input.MinPort,
+	}
+
+	return &output
+}
+
+func expandPortReuseHoldTimesModel(inputList []PortReuseHoldTimesModel) *attacheddatanetwork.PortReuseHoldTimes {
+	if len(inputList) == 0 {
+		return nil
+	}
+
+	input := &inputList[0]
+	output := attacheddatanetwork.PortReuseHoldTimes{
+		Tcp: &input.Tcp,
+		Udp: &input.Udp,
+	}
+
+	return &output
+}
+
+func flattenNaptConfigurationModel(input *attacheddatanetwork.NaptConfiguration) []NaptConfigurationModel {
+	var outputList []NaptConfigurationModel
+	if input == nil {
+		return outputList
+	}
+
+	output := NaptConfigurationModel{
+		Enabled: input.Enabled != nil && *input.Enabled == attacheddatanetwork.NaptEnabledEnabled,
+	}
+
+	if input.PinholeLimits != nil {
+		output.PinholeLimits = *input.PinholeLimits
+	}
+
+	output.PinholeTimeouts = flattenPinholeTimeoutsModel(input.PinholeTimeouts)
+
+	output.PortRange = flattenPortRangeModel(input.PortRange)
+
+	output.PortReuseHoldTime = flattenPortReuseHoldTimesModel(input.PortReuseHoldTime)
+
+	return append(outputList, output)
```

Comment:
Can we just combine this with L534 to be: `return []NaptConfigurationModel{output}`?